### PR TITLE
[M1386-2]

### PR DESCRIPTION
**Pull request by tienhoah**

### CODE REVIEW

**Comment by tienhoah** [comment]
[Collect - allow PIT transect length to have one decimal, e.g., 10.5m length (currently, integer only)](https://trello.com/c/OFDr06ry/1386-collect-allow-pit-transect-length-to-have-one-decimal-eg-105m-length-currently-integer-only)

File: src/api/ingest/fishbelt_serializer.py [hunk 1]
```diff
@@ -81,7 +81,8 @@ class FishBeltCSVSerializer(CollectRecordCSVSerializer):
         choices=tide_choices, required=False, allow_null=True, allow_blank=True
     )
 
-    data__fishbelt_transect__len_surveyed = serializers.IntegerField(min_value=0)
+    # data__fishbelt_transect__len_surveyed = serializers.IntegerField(min_value=0)
+    data__fishbelt_transect__len_surveyed = serializers.DecimalField(max_digits=4, decimal_places=1)
```
Comment (saanobhaai) [inline src/api/ingest/fishbelt_serializer.py:85]: Looks like you've done fishbelt and benthic PIT; but we should also do benthic LIT and habitatcomplexity

File: src/api/migrations/0008_auto_20200903_2034.py [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+    dependencies = [
+        ('api', '0007_auto_20200819_2130'),
+    ]
+
+    operations = [
+        migrations.RunSQL(model_view_migrations.reverse_sql(), model_view_migrations.forward_sql()),
+        migrations.AlterField(
+            model_name='benthictransect',
+            name='len_surveyed',
+            field=models.DecimalField(decimal_places=1, max_digits=4, validators=[django.core.validators.MinValueValidator(10), django.core.validators.MaxValueValidator(100)], verbose_name='transect length surveyed (m)'),
```
Comment (saanobhaai) [inline src/api/migrations/0008_auto_20200903_2034.py:19]: I think we shouldn't put model-level validators on this; submission validations should take care of warnings

File: src/api/models/mermaid.py [hunk 3]
```diff
@@ -499,8 +499,9 @@ def __str__(self):
 
 
 class Transect(SampleUnit):
-    len_surveyed = models.PositiveSmallIntegerField(
-        verbose_name=_(u'transect length surveyed (m)'))
+    len_surveyed = models.DecimalField(max_digits=4, decimal_places=1,
+                                    validators=[MinValueValidator(10), MaxValueValidator(100)],
```
Comment (saanobhaai) [inline src/api/models/mermaid.py:503]: See above -- I think we shouldn't put model-level validators on this

**Review by saanobhaai — CHANGES_REQUESTED** [review]

File: src/api/models/mermaid.py [hunk 4]
```diff
@@ -499,8 +499,8 @@ def __str__(self):
 
 
 class Transect(SampleUnit):
-    len_surveyed = models.PositiveSmallIntegerField(
-        verbose_name=_(u'transect length surveyed (m)'))
+    len_surveyed = models.DecimalField(max_digits=4, decimal_places=1,
```
Comment (saanobhaai) [inline src/api/models/mermaid.py:502]: fix spacing/alignment

File: src/api/ingest/benthicpit_serializer.py [hunk 5]
```diff
@@ -80,7 +80,8 @@ class BenthicPITCSVSerializer(CollectRecordCSVSerializer):
 
     data__interval_size = serializers.DecimalField(max_digits=4, decimal_places=2)
     data__interval_start = serializers.DecimalField(max_digits=4, decimal_places=2)
-    data__benthic_transect__len_surveyed = serializers.IntegerField(min_value=0)
+    # data__benthic_transect__len_surveyed = serializers.IntegerField(min_value=0)
```
Comment (saanobhaai) [inline src/api/ingest/benthicpit_serializer.py:83]: delete commented line here and below

**Review by saanobhaai — CHANGES_REQUESTED** [review]
@tienhoah all looks good; just a couple of cleanup comments. Then I'll likely update the migrations based on others that have happened in the meantime.

**Pull request merged**